internal/extract/getter.go

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package extract
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+	"go.uber.org/zap"
+)
+
+type getter struct {
+	logger *zap.Logger
+}
+
+func newGetter(options ...GetterOption) *getter {
+	getter := &getter{
+		logger: zap.NewNop(),
+	}
+	for _, option := range options {
+		option(getter)
+	}
+	return getter
+}
+
+func (g *getter) GetField(fileDescriptorSets []*descriptor.FileDescriptorSet, path string) (*Field, error) {
+	if len(path) == 0 {
+		return nil, fmt.Errorf("empty path")
+	}
+	if path[0] == '.' {
+		path = path[1:]
+	}
+	split := strings.Split(path, ".")
+	if len(split) < 2 {
+		return nil, fmt.Errorf("no field for path %s", path)
+	}
+	message, err := g.GetMessage(fileDescriptorSets, strings.Join(split[0:len(split)-1], "."))
+	if err != nil {
+		return nil, err
+	}
+	var foundFieldDescriptorProto *descriptor.FieldDescriptorProto
+	for _, fieldDescriptorProto := range append(message.GetField(), message.GetExtension()...) {
+		if fieldDescriptorProto.GetName() == split[len(split)-1] {
```

Comment:
nit: maybe capture `split[len(split)-1]` as `wantName` or something for better readability

Reply:
Sure why not :-)